include/lkmm.h

```diff
@@ -1,45 +1,59 @@
 #include <stdint.h>
 
-typedef enum memory_order {
-  memory_order_relaxed,
-  memory_order_once,
-  memory_order_acquire,
-  memory_order_release,
-  mb,
-  wmb,
-  rmb,
-  rcu_lock,
-  rcu_unlock,
-  rcu_sync,
-  before_atomic,
-  after_atomic,
-  after_spinlock,
-  barrier,
-} memory_order;
-
-typedef enum operation {
-  op_add,
-  op_sub,
-  op_and,
-  op_or,
-} operation;
-
-extern int __LKMM_LOAD(void*, memory_order);
-extern void __LKMM_STORE(void*, int, memory_order);
-extern void __LKMM_FENCE(int);
-extern int __LKMM_XCHG(int*, int, memory_order);
-extern int __LKMM_CMPXCHG(int*, int, int, memory_order, memory_order);
-extern void __LKMM_ATOMIC_OP(int*, int, operation);
-extern int __LKMM_ATOMIC_FETCH_OP(int*, int, memory_order, operation);
-extern int __LKMM_ATOMIC_OP_RETURN(int*, int, memory_order, operation);
+typedef enum __LKMM_memory_order {
+  __LKMM_relaxed,
+  __LKMM_once,
+  __LKMM_acquire,
+  __LKMM_release,
+  __LKMM_mb,
+  __LKMM_wmb,
+  __LKMM_rmb,
+  __LKMM_rcu_lock,
+  __LKMM_rcu_unlock,
+  __LKMM_rcu_sync,
+  __LKMM_before_atomic,
+  __LKMM_after_atomic,
+  __LKMM_after_spinlock,
+  __LKMM_barrier,
+} __LKMM_memory_order;
+
+typedef enum __LKMM_operation {
+  __LKMM_op_add,
+  __LKMM_op_sub,
+  __LKMM_op_and,
+  __LKMM_op_or,
+} __LKMM_operation;
+
+typedef typeof(sizeof(int)) __LKMM_size_t;
+typedef long long __LKMM_int_t;
```

Comment:
Why has this become a 64 bit int?

Reply:
It is supposed to cover the signed and unsigned value space of the maximal supported value type.  It could be `intmax_t`.